monai/data/image_reader.py

```diff
@@ -740,19 +732,36 @@ def get_data(
     def _extract_region(
         self,
         img_obj,
-        size: Tuple[int, int],
+        size: Optional[Tuple[int, int]],
         location: Tuple[int, int] = (0, 0),
         level: int = 0,
         dtype: DtypeLike = np.uint8,
     ):
         # reverse the order of dimensions for size and location to be compatible with image shape
-        size = size[::-1]
         location = location[::-1]
-        region = img_obj.read_region(location=location, size=size, level=level)
+        if size is None:
+            region = img_obj.read_region(location=location, level=level)
```

Comment:
could you please test this

Reply:
It is already being tested by existing unittests: https://github.com/Project-MONAI/MONAI/blob/9f6ceebca87c06ec9fed8447791c9a7f3378ef0a/tests/test_cuimage_reader.py#L71

Reply:
It is already being tested by existing unittests: https://github.com/Project-MONAI/MONAI/blob/9f6ceebca87c06ec9fed8447791c9a7f3378ef0a/tests/test_cuimage_reader.py#L71

Reply:
@wyli, it this what you meant? When it is called by `img_obj` only, it invokes `region = img_obj.read_region(location=(0, 0), level=0)`.

Reply:
exactly, when size is None

Reply:
this change would break the openslide reader, e.g. test case:
```
{    {
        "location": (0, 0),
        "size": None,
        "level": 2,
        "grid_shape": (2, 1),
        "patch_size": 2,
    },
```
outcome:
```
======================================================================
ERROR: test_read_whole_image_0__opt_monai_tests_testing_data_CMU_1_tiff (__main__.TestOpenSlideReader)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/site-packages/parameterized/parameterized.py", line 533, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "/opt/monai/tests/test_openslide_reader.py", line 75, in test_read_whole_image
    img = reader.get_data(img_obj)[0]
  File "/opt/monai/monai/data/image_reader.py", line 716, in get_data
    region = self._extract_region(img, location=location, size=size, level=level, dtype=dtype)
  File "/opt/monai/monai/data/image_reader.py", line 743, in _extract_region
    region = img_obj.read_region(location=location, level=level)
TypeError: read_region() missing 1 required positional argument: 'size'
```

Reply:
Oh, that's right but it didn't break initially. Maybe I have changed something again. I'll fix it.